libslide/src/parser.rs

```diff
@@ -198,6 +188,30 @@ where
 
         node
     }
+
+    /// Creates diagnostics for extra tokens following a primary item.
+    /// All remaining tokens will be consumed in the construction of the diagnostic.
+    ///
+    /// `additional_diags` applies additional diagnostic messages to the extra tokens diagnostic,
+    /// if one is produced.
+    fn extra_tokens_diag(&mut self, additional_diags: impl Fn(Diagnostic, Span) -> Diagnostic) {
+        while matches!(self.peek().ty, TT::CloseParen | TT::CloseBracket) {
```

Comment:
i dont understand while if it does match, we push unmatched delim

Reply:
While we have an unmatched delim, add an error for that.

```
1+ )]) asdasd
   ^ unmatched delim
    ^ unmatched delim
     ^ unmatched delim
       ^^^^^^ extra tokens
```